mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -557,6 +604,180 @@ class WidgetAdapter(
         }
     }
 
+    class InputViewHolder internal constructor(initData: ViewHolderInitData) :
+        LabeledItemBaseViewHolder(initData, R.layout.widgetlist_inputitem, R.layout.widgetlist_inputitem_compact) {
+        private val inputTextLayout: TextInputLayout = itemView.findViewById(R.id.widgetinput)
+        private val inputText: TextInputEditText = itemView.findViewById(R.id.widgetinputvalue)
+        private var isBinding = false
+        private var hasChanged = false
+        private var showingDatePicker = false
+        private var showingTimePicker = false
+
+        private var updateJob: Job? = null
+        private var oldValue: String? = null
+
+        init {
+            inputText.doAfterTextChanged {
+                if (!isBinding) {
+                    hasChanged = true
+                }
+            }
+            inputText.setOnFocusChangeListener { _, hasFocus ->
+                if (!hasFocus && hasChanged) {
+                    updateValue()
+                }
+            }
+        }
+
+        override fun bind(widget: Widget) {
+            isBinding = true
+            updateJob?.cancel()
+
+            super.bind(widget)
+
+            val item = widget.item
+
+            inputText.inputType = when (widget.inputHint) {
+                Widget.Hint.Number -> (TYPE_CLASS_NUMBER or TYPE_NUMBER_FLAG_DECIMAL or TYPE_NUMBER_FLAG_SIGNED)
+                Widget.Hint.Date -> (TYPE_CLASS_DATETIME or TYPE_DATETIME_VARIATION_DATE)
+                Widget.Hint.Time -> (TYPE_CLASS_DATETIME or TYPE_DATETIME_VARIATION_TIME)
+                Widget.Hint.Datetime -> (TYPE_CLASS_DATETIME or TYPE_DATETIME_VARIATION_NORMAL)
+                else -> TYPE_CLASS_TEXT
+            }
+
+            val displayState = widget.stateFromLabel?.replace("\n", "") ?: ""
+
+            inputTextLayout.placeholderText = if (widget.state == null) {
+                if (
+                    (widget.inputHint == Widget.Hint.Text) and
+                    (item?.isOfTypeOrGroupType(Item.Type.DateTime) == true)
+                ) {
+                    "YYYY-MM-DD hh:mm:ss"
+                } else displayState
+            } else ""
+
+            val dataState = if (widget.state != null) {
+                if (widget.inputHint == Widget.Hint.Number)
+                    widget.state.asNumber?.formatValue()
+                else if (displayState.isNotEmpty()) displayState
+                else if (widget.item?.isOfTypeOrGroupType(Item.Type.DateTime) == true)
+                    when (widget.inputHint) {
+                        Widget.Hint.Date -> widget.state.asDateTime?.toISOLocalDate()
+                        Widget.Hint.Time -> widget.state.asDateTime?.toISOLocalTime()
```

Comment:
Wouldn't a locale dependent format be nicer/better here?

Reply:
Yes, potentially. I will check that out.

Reply:
Done.